pkg/controller/realmadmin/save.go

```diff
@@ -48,14 +56,74 @@ func (c *Controller) HandleSave() http.Handler {
 			return
 		}
 
+		// SMS config is all or nothing
+		if (form.TwilioAccountSid != "" || form.TwilioAuthToken != "" || form.TwilioFromNumber != "") &&
+			(form.TwilioAccountSid == "" || form.TwilioAuthToken == "" || form.TwilioFromNumber == "") {
+			flash.Error("Error updating realm: all SMS fields are required")
```

Comment:
nit: not quite accurate, either they're all required or must be blank to disable SMS

Reply:
Done